src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextHost.WorkspaceProjectContextHostInstance.cs

```diff
@@ -143,16 +144,21 @@ private async Task ApplyProjectChangesUnderLockAsync(IProjectVersionedValue<IPro
                     {
                         await _applyChangesToWorkspaceContext.Value.ApplyProjectBuildAsync(update, isActiveContext, cancellationToken);
                     }
+
+                    context.EndBatch();
+                    await _applyChangesToWorkspaceContext.Value.ApplyProjectUpdatedAsync(update, isActiveContext, cancellationToken);
```

Comment:
Why can't this just be moved outside of the try...finally to avoid keeping track if the batch has finished?